static/src/js/components/Permission/Permission.jsx

```diff
@@ -0,0 +1,283 @@
+    }
+
+    return {
+      name: groupName,
+      id,
+      search: permissions.includes('read'),
+      order: permissions.includes('order')
+    }
+  })
+
+  // Builds text for access constraints
+  const buildAccessConstraintText = (accessValue) => {
+    const {
+      includeUndefinedValue,
+      maxValue,
+      minValue
+    } = accessValue || {}
+
+    const accessFragment = [' that have access constraint values']
+
+    if (minValue && maxValue) {
+      if (minValue === maxValue) {
+        accessFragment.push(` equal to ${minValue}`)
+      } else {
+        accessFragment.push(` between ${minValue} and ${maxValue}`)
+      }
+
+      if (includeUndefinedValue) {
+        accessFragment.push(' (or are undefined)')
+      }
+    }
+
+    return accessFragment
+  }
+
+  // Builds text for temporal constraints
+  const buildTemporalText = (temporalObj) => {
+    const {
+      mask,
+      startDate,
+      stopDate
+    } = temporalObj
+
+    const temporalFragment = [' that have a start and end date']
+
+    switch (mask) {
+      case 'contains':
+        temporalFragment.push(' contained within')
+        break
+      case 'intersect':
+        temporalFragment.push(' having some overlap with')
+        break
+      case 'disjoint':
+        temporalFragment.push(' outside of')
+        break
+      default: break
+    }
+
+    temporalFragment.push(` the date range ${startDate} to ${stopDate}`)
+
+    return temporalFragment
+  }
+
+  const buildAccessAndTemporalText = (accessValue, temporalObj) => {
+    const sentenceFragments = []
+    if (accessValue && temporalObj) {
+      sentenceFragments.push(buildAccessConstraintText(accessValue), ' and', buildTemporalText(temporalObj))
+
+      return sentenceFragments
+    }
+
+    if (accessValue) {
+      sentenceFragments.push(buildAccessConstraintText(accessValue))
+    }
+
+    if (temporalObj) {
+      sentenceFragments.push(buildTemporalText(temporalObj))
+    }
+
+    return sentenceFragments
+  }
+
+  const collectionConstraintSummary = () => {
+    const sentenceFragments = ['This permission']
+
+    if (collectionApplicable) {
+      if (collectionCount) {
+        sentenceFragments.push(` grants its assigned groups access to ${collectionCount} collection${collectionCount > 1 ? 's' : ''}`)
+      } else {
+        sentenceFragments.push(' grants its assigned groups access to all of its collections')
+      }
+
+      sentenceFragments.push(buildAccessAndTemporalText(collectionAccessValue, temporal))
+    } else {
+      sentenceFragments.push(' does not grant access to collections')
+    }
+
+    return sentenceFragments
+  }
+
+  const granuleConstraintSummary = () => {
+    const {
+      accessValue: granuleAccessValue,
+      temporal: granuleTemporal = null
+    } = granuleIdentifier || {}
+
+    const sentenceFragments = ['This permission']
+
+    if (granuleApplicable) {
+      sentenceFragments.push(' grants its assigned groups access to granules')
```

Comment:
Unless there is some kind of punctuation you're dealing with, might be best to add these strings without leading spaces and join them all at the end with `' '`